docs/MarketingCloud-batchsource.md

```diff
@@ -0,0 +1,96 @@
+
+`Unsub Event` - will allow user to fetch Tracking UnSubscribe Events,
+
+`Sent Event` - will allow user to fetch Tracking Sent Events,
+
+`UnSent Event` - will allow user to fetch Tracking UnSent Events.
+
+Note, this value will be ignored if the Mode is set to `Multi Object`.  
+
+**Data Extension External Key**: Specify the data extension key from which data to be fetched. Note, this value will 
+be ignored in following two cases: 
+
+* If the Mode is set to `Multi Object`
+* If the selected object name is other than `Data Extension`. 
+
+**Object List**: Specify the comma-separated list of objects for which data to be fetched; for example: 
+'Object1,Object2'. This can be one or more values from following possible values: 
+
+`Data Extension` - will allow user to fetch data for a single Data Extension object,
+
+`Email` - will allow user to fetch data for Email object,
+
+`Mailing List` - will allow user to fetch data for Mailing List object
+
+`Bounce Event` - will allow user to fetch Tracking Bounce Events,
+
+`Open Event` - will allow user to fetch Tracking Open Events,
+
+`Unsub Event` - will allow user to fetch Tracking UnSubscribe Events,
+
+`Sent Event` - will allow user to fetch Tracking Sent Events,
+
+`NotSent Event` - will allow user to fetch Tracking Not Sent Events.
+
+Note, this value will be ignored if the Mode is set to `Single Object`.
+
+**Data Extension External Keys**: Specify the data extension keys from which data to be fetched; for example: 
+'Key1,Key2'. Note, this value will be ignored in following two cases: 
+
+* If the Mode is set to `Single Object`,
+
+* If the selected object list does not contain `Data Extension` as one of the objects.
+
+**Table Name Field**: The name of the field that holds the object name to which the data belongs to. Must not be the 
+name of any column for any of the objects that will be read. Defaults to `tablename`. In case of `Data Extension` 
+object, this field will have value in `dataextension_[Data Extension Key]` format. Note, the Table name field value 
+will be ignored if the Mode is set to `Single Object`.
+
+**Filter**: The filter selection criteria. eg: eventDate > '2010-01-01' and subscriberKey = 1001
+
+**Client ID**: OAuth2 client ID associated with an installed package in the Salesforce Marketing Cloud.
+
+**Client Secret**: TOAuth2 client secret associated with an installed package in the Salesforce Marketing Cloud.
+
+**Authentication Base URI**: Authentication Base URL associated for the Server-to-Server API integration. 
+For example, `https://instance.auth.marketingcloudapis.com/`
+
+**SOAP API Endpoint**: The SOAP Endpoint URL associated for the Server-to-Server API integration. For example, 
+`https://instance.soap.marketingcloudapis.com/Service.asmx`
+
+**REST API Base URI**: The REST API Base URL associated for the Server-to-Server API integration. For example, 
+`https://instance.rest.marketingcloudapis.com/`
```

Comment:
this doc should have a mapping of SFMC data types to CDAP data types, similar to what is in https://github.com/data-integrations/google-cloud/blob/develop/docs/BigQueryTable-batchsink.md#data-type-mappings-from-cdap-to-bigquery